### ISSUE TITLE
astro-compress middleware

### ISSUE THREAD

**Issue by kireerik** [PlayForm/Compress#131]
> May make `on` (`true`) as the default setting in development mode as well for this new feature.

&nbsp;
related issue:
- #76

**Comment by nikolaxhristov** [comment]
Yeah, you can do:

```ts
import compress from "astro-compress";
import { defineConfig } from "astro/config";

export default defineConfig({
	integrations: [
		import.meta.env.MODE === "production" ? compress() : null,
	],
});
```

But the integration is not enabled in development it's only turned on [`"astro:build:done"`](https://docs.astro.build/en/reference/integrations-reference/#astrobuilddone)

**Comment by nikolaxhristov** [comment]
Apologies you meant, `on`. There is currently no way to turn it on in development mode as it served as a server from `vite`, but there are plans to make it integrate with the new middleware API https://docs.astro.build/en/guides/middleware/. Expect it in `1.2.0`

**Comment by kireerik** [comment]
Yes #76 is confusing for me. Why do @rdwz want to turn it off in development mode when it is not turned on anyway?

**Comment by NikolaRHristov** [comment]
It's useful to know that you can turn off an integration like that, but it's not specific to `astro-compress`

**Comment by nikolaxhristov** [comment]
Or maybe they have an environment where `import.meta.env.MODE` is always `development`. I don't know if this is affected by `process.env`

**Comment by kireerik** [comment]
But he writes `compress({html: false})` for development, which is currently the same as doing `compress({html: true})` or `compress()`, because for some reason in development, it is basically `false` regardless for some reason..

**Comment by nikolaxhristov** [comment]
Just a preference I think, but if you put `false` in any of the options it should not trigger compression for that specific file type.

**Comment by nikolaxhristov** [comment]
Yes, but that is regardless of the development mode, if you put `true` it will just trigger the compression with the defaults from https://github.com/astro-community/astro-compress/tree/main/src/options

**Comment by nikolaxhristov** [comment]
Exactly!

**Comment by nikolaxhristov** [comment]
The current way it works is just runs after build is done over all files.

**Comment by nikolaxhristov** [comment]
The new way (extra) would be to attach a middleware to an internal function inside `astro-compress`, so that every request gets served compressed, but the file stays unchanged

**Comment by kireerik** [comment]
I see. Nice. Can you link that issue in your backlog?

**Comment by nikolaxhristov** [comment]
Sure. 😁

**Comment by kireerik** [comment]
So can you link that issue in your backlog?

**Comment by NikolaRHristov** [comment]
@kireerik It's already linked, you'll get notified once it's done. 😁 The backlog is my personal task list.
![image](https://github.com/astro-community/astro-compress/assets/90222871/4e3950e1-bcaf-4148-847c-4f2d318e6238)
